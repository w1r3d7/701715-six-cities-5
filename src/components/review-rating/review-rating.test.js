import React from 'react';
import renderer from 'react-test-renderer';

import ReviewRating from './review-rating';

const title = `perfect`;
const rating = `5`;

describe(`ReviewRating render correctly`, () => {
  it(`Disable ReviewRating`, () => {
    const tree = renderer
      .create(
          <ReviewRating
            rating={rating}
            isDisabled={true}
            title={title}
          />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Enable ReviewRating`, () => {
    const tree = renderer
      .create(
          <ReviewRating
            rating={rating}
            isDisabled={false}
            title={title}
          />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

