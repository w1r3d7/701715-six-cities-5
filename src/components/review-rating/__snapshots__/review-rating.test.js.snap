// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewRating render correctly Disable ReviewRating 1`] = `
Array [
  <input
    className="form__rating-input visually-hidden"
    disabled="disabled"
    id="5-stars"
    name="rating"
    required={true}
    type="radio"
    value="5"
  />,
  <label
    className="reviews__rating-label form__rating-label"
    htmlFor="5-stars"
    title="perfect"
  >
    <svg
      className="form__star-image"
      height="33"
      width="37"
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
  </label>,
]
`;

exports[`ReviewRating render correctly Enable ReviewRating 1`] = `
Array [
  <input
    className="form__rating-input visually-hidden"
    disabled=""
    id="5-stars"
    name="rating"
    required={true}
    type="radio"
    value="5"
  />,
  <label
    className="reviews__rating-label form__rating-label"
    htmlFor="5-stars"
    title="perfect"
  >
    <svg
      className="form__star-image"
      height="33"
      width="37"
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
  </label>,
]
`;

exports[`ReviewRating render correctly Not disable ReviewRating 1`] = `
Array [
  <input
    className="form__rating-input visually-hidden"
    disabled=""
    id="5-stars"
    name="rating"
    required={true}
    type="radio"
    value="5"
  />,
  <label
    className="reviews__rating-label form__rating-label"
    htmlFor="5-stars"
    title="perfect"
  >
    <svg
      className="form__star-image"
      height="33"
      width="37"
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
  </label>,
]
`;
